multiqc/modules/haplocheck/haplocheck.py

```diff
@@ -0,0 +1,137 @@
+            ),
+            doi="10.1101/gr.256545.119",
+        )
+        log_files = self.find_log_files("haplocheck", filehandles=True)
+        samples = {}
+        for f in log_files:
+            samples = self._parse_samples(f, samples)
+            self.add_data_source(f)
+
+        self._samples = samples
+
+        self._add_table(samples)
+        self._add_coverage_plot(samples)
+        self._add_heteroplasmy_level_plot(samples)
+
+        self.write_data_file(self._samples, "multiqc_haplocheck")
+        log.info("Found {} samples".format(len(samples)))
+
+    def _add_table(self, samples):
+        self.add_section(
+            description=(
+                "For further information regarding the output values, refer to the "
+                "[tool's documentation](https://mitoverse.readthedocs.io/haplocheck/haplocheck/#textual-report-file)."
+            ),
+            plot=table.plot(data=samples, headers=self._setup_headers()),
+        )
+
+    def _add_coverage_plot(self, samples):
+        coverage_plot = bargraph.plot(
+            data=samples,
+            cats=["Sample Coverage"],
+            pconfig={
+                "id": "haplocheck-coverage",
+                "title": "Haplocheck: coverages",
+                "xlab": "Sample",
+                "ylab": "Average Coverage",
+                "tt_suffix": "x",
+            },
+        )
+        self.add_section(
+            name="Average coverage per sample",
+            anchor="haplocheck-coverage-id",
+            description="Average coverage per sample",
+            plot=coverage_plot,
+        )
+
+    def _add_heteroplasmy_level_plot(self, samples):
+        coverage_plot = bargraph.plot(
+            data=samples,
+            cats=["Major Heteroplasmy Level", "Minor Heteroplasmy Level"],
+            pconfig={
+                "id": "haplocheck-het-level",
+                "title": "Haplocheck: Heteroplasmy Level",
+                "xlab": "Sample",
+                "ylab": "Heteroplasmy Level",
+                "tt_decimals": 2,
+            },
+        )
+        self.add_section(
+            name="Heteroplasmy level per sample",
+            anchor="haplocheck-het-level-id",
+            description="Heteroplasmy level per sample",
+            plot=coverage_plot,
+        )
+
+    def _setup_headers(self):
+        headers = OrderedDict()
+        for k in list(self._samples.values())[0]:
+            headers[k] = {"title": k, "hidden": True}
+
+        headers["Contamination Status"] = {
+            "title": "Contamination Status",
+            "description": "This column can either be YES, NO or ND (not detectable)",
+            "hidden": False,
+            "bgcols": {"YES": "#f8d7da", "ND": "#fff3cd", "NO": "#d1e7dd"},
+        }
+        headers["Contamination Level"] = {
+            "title": "Contamination Level",
+            "description": "The detected contamination level for the sample",
+            "bgcols": {"ND": "#d1e7dd"},
```

Comment:
It's a bit confusing that `ND` here has the same color as `NO` above, and `ND` above is different. 
What are other values that "Contamination Level" can take?

Reply:
Do you think these two columns can be combined? Like in, a column would either show a `Not detectable`, or a number corresponding to the contamination level, with 0 meaning zero contamination. That one metric can be also shared in the general stats.